### PR TITLE
Add .kind prop to Link. (supports `primary` and `secondary` kind)

### LINKED ISSUE #97: [UI core library] Secondary (footer) link type
We need to implement secondary link type for Cloud (footer links on design below):
https://www.figma.com/file/kaC3jgqMSgqMEgnv7TIse1/%F0%9F%93%90Sign-in-flow?node-id=1350%3A1962

### PR DESCRIPTION
**Pull request by utkukaratas**
Closes #97

### CODE REVIEW

File: packages/ui/src/Link.tsx
Comment (aigoncharov) [inline packages/ui/src/Link.tsx:19]: nit: Why do we need a new type? Could we just add a `kind` prop to `LinkProps` next to `size`?

File: packages/ui/__stories__/Link.stories.tsx
Comment (aigoncharov) [inline packages/ui/__stories__/Link.stories.tsx:106]: Could you generate screenshots for these stories? `npm run generate-screenshots`

File: packages/ui/src/Link.tsx
Comment (utkukaratas) [reply packages/ui/src/Link.tsx:19]: just imitated the Button comp.

Comment (aigoncharov) [reply packages/ui/src/Link.tsx:19]: For Button we do this because we have an intersection with several other union types  - ButtonAccessibleIconLeftProps, ButtonAccessibleIconRightProps. This component is much simpler. WDYT?

File: packages/ui/__stories__/Link.stories.tsx
Comment (utkukaratas) [reply packages/ui/__stories__/Link.stories.tsx:106]: @aigoncharov how about now? (sorry bit of a foreign workflow for me, will take some time to adapt ;))

File: packages/ui/src/Link.tsx
Comment (utkukaratas) [reply packages/ui/src/Link.tsx]: no favorites for me. (my type game will also take some time; you know a decade of duck typing rots one's brain :)) so the last commit is more preferable you think?

**Review by danad02 — APPROVED** [review]
LGTM :+1:

**Review by aigoncharov — APPROVED** [review]
LGTM!

**Review by jvorcak — APPROVED** [review]

**Comment by aigoncharov** [comment]
@utkukaratas PR builder is failing for some reason. Could you try `npm run lint` and `npm test` locally?

**Comment by utkukaratas** [comment]
npm test/lint work fine locally. did a rebase and pushed again.

**Comment by aigoncharov** [comment]
@utkukaratas visual regression test suite fails on Jenkins 
```
15:29:28 @hazelcast/ui: FAIL chrome.docker/chrome.laptop/Components/Link/Secondary Normal With Icon: Screenshot differs from reference, see .loki/difference/chrome_laptop_Components_Link_Secondary_Normal_With_Icon.png
15:29:28 @hazelcast/ui: RUNS chrome.docker/chrome.laptop/components/NumberBadge/Less Than 100
15:29:29 @hazelcast/ui: PASS chrome.docker/chrome.laptop/Components/Link/Secondary Small
15:29:29 @hazelcast/ui: RUNS chrome.docker/chrome.laptop/components/NumberBadge/More Than 99
15:29:29 @hazelcast/ui: FAIL chrome.docker/chrome.laptop/Components/Link/Secondary Small With Icon: Screenshot differs from reference, see .loki/difference/chrome_laptop_Components_Link_Secondary_Small_With_Icon.png
```
BTW, could you ask Emre where you could get the VPN credentials to access jenkins?

**Comment by utkukaratas** [comment]
Right now visual tests fail "Timeout after 10000ms". That's weird. Any have hints for the reason? @aigoncharov

**Comment by aigoncharov** [comment]
> Right now visual tests fail "Timeout after 10000ms". That's weird. Any have hints for the reason? @aigoncharov

@utkukaratas I have never seen it

**Comment by utkukaratas** [comment]
okay, will try on my beefier home desktop (luckily the laptop which i'm currently on might be the culprit).

**Comment by aigoncharov** [comment]
@utkukaratas did you run `npm run build-storybook` before running `npm run test:visual`? I also had some weird hanging until I built the storybook.

**Comment by utkukaratas** [comment]
fixed my fuckup. all tests pass. will merge once approved.

**Comment by aigoncharov** [comment]
@utkukaratas congrats on you first PR to `frontend-shared`!

**Pull request merged**
